src/amo/sagas/addonsByAuthors.js

```diff
@@ -0,0 +1,45 @@
+import { call, put, select, takeEvery } from 'redux-saga/effects';
+import { SEARCH_SORT_POPULAR } from 'core/constants';
+import {
+  FETCH_OTHER_ADDONS_BY_AUTHORS,
+  NB_OTHERS_ADDONS_BY_AUTHORS,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import searchApi from 'core/api/search';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+
+export function* fetchOtherAddonsByAuthors({ payload }) {
+  const { errorHandlerId, authors, slug, addonType } = payload;
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const response = yield call(searchApi, {
+      api: state.api,
+      filters: {
+        addonType,
+        author: authors.join(','),
+        // We need one more add-on than the number to display because the API
+        // may return the main add-on and we cannot tell the API to exclude it.
+        page_size: NB_OTHERS_ADDONS_BY_AUTHORS + 1,
+        sort: SEARCH_SORT_POPULAR,
+      },
+    });
+
+    const addons = Object.values(response.entities.addons || {});
```

Comment:
This seems like something the reducer should be doing, not the saga.

Reply:
Then it is not clear to me what goes in saga and what not.

For instance here, without Normalizr I would pass `response.addons` and I would not have to change any line of code in the reducer. Yet we have this weird Normalizr layer that changes the response payload, transforms a simple array into a hash map, and does not return anything when array is empty (reported by @kumar303 yesterday).

Reply:
Oh, right. Okay, as long as we have normalizr (which, I agree, is annoying) this makes sense. I guess we could add a TODO here that this can be removed once we ditch normalizr (#2917).